### PR TITLE
Improve compatibility with MetaMask and allow web3 provider selection

### PR DESCRIPTION
**Pull request by bbondy**
Deprecate enabled setting in favour of lazy loading on demand if None is
set as the provider.

Fix https://github.com/brave/brave-browser/issues/7503

See the spec in the issue for details of how this all works.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [x] iOS
  - [x] Linux
  - [x] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [x] iOS
  - [x] Linux
  - [x] macOS
  - [x] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [x] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/brave_wallet/brave_wallet_utils.cc [hunk 1]
```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
```
Comment (emerick) [inline browser/brave_wallet/brave_wallet_utils.cc]: Nit: Might want to update these to 2020 now.

File: browser/resources/settings/brave_default_extensions_page/brave_default_extensions_page.js [hunk 2]
```diff
@@ -56,6 +56,10 @@ Polymer({
     this.browserProxy_.isTorManaged().then(managed => {
       this.disableTorOption_ = managed
     })
+    this.browserProxy_.getWeb3ProviderList().then(list => {
+      this.braveWeb3Providers_ = JSON.parse(list);
+      console.log("found provider list JSON of: ", list)
```
Comment (emerick) [inline browser/resources/settings/brave_default_extensions_page/brave_default_extensions_page.js]: Leaving this in intentionally?

File: components/brave_extension/extension/brave_extension/background/reducers/dappDetectionReducer.ts [hunk 3]
```diff
@@ -9,17 +9,16 @@ export default function dappDetectionReducer (state = {}, action: Actions) {
...
-              return
-            }
-            chrome.tabs.executeScript(action.tabId, {
-              file: 'out/content_dapps.bundle.js',
-              allFrames: false,
-              runAt: 'document_start',
-              frameId: 0
-            })
+        chrome.braveWallet.shouldCheckForDapps((dappDetection) => {
+          console.log('shouldcheckfor dapps: ', dappDetection)
```
Comment (emerick) [inline components/brave_extension/extension/brave_extension/background/reducers/dappDetectionReducer.ts]: Leaving this in intentionally?

File: browser/brave_wallet/brave_wallet_utils.cc (hunk 1 above)
Comment (bbondy) [reply browser/brave_wallet/brave_wallet_utils.cc]: updated all new instances of 2019 to 2020. 👍

File: components/brave_extension/extension/brave_extension/background/reducers/dappDetectionReducer.ts (hunk 3 above)
Comment (bbondy) [reply components/brave_extension/extension/brave_extension/background/reducers/dappDetectionReducer.ts]: Not intentional, thx, removed.

File: browser/resources/settings/brave_default_extensions_page/brave_default_extensions_page.js (hunk 2 above)
Comment (bbondy) [reply browser/resources/settings/brave_default_extensions_page/brave_default_extensions_page.js]: Not intentional, thx. Removed.

**Pull request merged**